anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/source/data/initializer/DataInitializer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.source.data.initializer;
+
+import com.expedia.adaptivealerting.anomdetect.detect.Detector;
+import com.expedia.adaptivealerting.anomdetect.detect.MappedMetricData;
+import com.expedia.adaptivealerting.anomdetect.detect.outlier.algo.forecasting.ForecastingDetector;
+import com.expedia.adaptivealerting.anomdetect.forecast.point.algo.seasonalnaive.SeasonalNaivePointForecaster;
+import com.expedia.adaptivealerting.anomdetect.source.data.DataSource;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+@AllArgsConstructor
+@Slf4j
+public class DataInitializer {
+    //TODO Find appropriate earliest time and max data point values.
+    private static final String EARLIEST_TIME = "7d";
+    private static final Integer MAX_DATA_POINTS = 2016;
+
+    @NonNull
+    private DataSource dataSource;
+
+    public void initializeDetector(MappedMetricData metricData, Detector detector) {
+        if (detector != null && "seasonalnaive".equals(detector.getName())) {
+            val forecastingDetector = (ForecastingDetector) detector;
+            val seasonalNaivePointForecaster = (SeasonalNaivePointForecaster) forecastingDetector.getPointForecaster();
```

Comment:
We should avoid casting to specific `SeasonalNaivePointForecaster` class. We should only need to use the `forecastingDetector.forecast()` method.

Something like this:

```
    public void initializeDetector(MappedMetricData mappedMetricData, Detector detector) {
        if (detector != null) {
            if (detector instanceof SeasonalNaivePointForecaster) {
                val data = getHistoricalData(mappedMetricData);
                val forecastingDetector = (ForecastingDetector) detector;
                MetricDefinition metricDefinition = mappedMetricData.getMetricData().getMetricDefinition();
                populateForecastingDetectorWIthHistoricalData(forecastingDetector, data, metricDefinition);
            }
        }
    }

    private List<DataSourceResult> getHistoricalData(MappedMetricData mappedMetricData) {
        val target = getTarget(mappedMetricData);
        return dataSource.getMetricData(EARLIEST_TIME, MAX_DATA_POINTS, target);
    }

    // TODO: Move this to separate class
    private String getTarget(MappedMetricData mappedMetricData) {
        val metricData = mappedMetricData.getMetricData();
        val metricDefinition = metricData.getMetricDefinition();
        val metricTags = metricDefinition.getTags();
        val metricFunction = metricTags != null && metricTags.getKv() != null
                ? metricTags.getKv().get("function") : null;
        return metricFunction != null
                ? metricFunction : metricDefinition.getKey();
    }

    private void replayHistoricalDataForForecastingDetector(ForecastingDetector forecastingDetector, List<DataSourceResult> data, MetricDefinition metricDefinition) {
        for (DataSourceResult dataSourceResult : data) {
            MetricData metricData = dataSourceResultToMetricData(dataSourceResult, metricDefinition);
            // We throw away the forecast as we only care about letting the forecaster "see" the historical value to update its internal model
            forecastingDetector.getPointForecaster().forecast(metricData);
        }
    }

    // TODO: Move this to a separate class
    private MetricData dataSourceResultToMetricData(DataSourceResult dataSourceResult, MetricDefinition metricDefinition) {
        Double dataPoint = dataSourceResult.getDataPoint();
        long epochSecond = dataSourceResult.getEpochSecond();
        return new MetricData(metricDefinition, dataPoint, epochSecond);
    }

```